mila_datamodules/cli/prepare_torchvision.py

```diff
@@ -146,56 +256,286 @@ def __call__(
+                # dataset
+                files=[
+                    ("train.tar.gz", "2021_train.tgz"),
+                    ("train_mini.tar.gz", "2021_train_mini.tgz"),
+                    ("val.tar.gz", "2021_valid.tgz"),
+                ]
+            ),
+            PrepareVisionDataset(tvd.INaturalist)
+        ])
+        for cluster, datasets_folder in standardized_torchvision_datasets_dirs.items()
     },
     tvd.ImageNet: {
         # TODO: Write a customized `PrepareVisionDataset` for ImageNet that uses Olexa's magic tar
         # command.
-        cluster: SymlinkArchives(
-            tvd.ImageNet,
-            {
-                "ILSVRC2012_devkit_t12.tar.gz": f"{folder}/imagenet/ILSVRC2012_devkit_t12.tar.gz",
-                "ILSVRC2012_img_train.tar": f"{folder}/imagenet/ILSVRC2012_img_train.tar",
-                "ILSVRC2012_img_val.tar": f"{folder}/imagenet/ILSVRC2012_img_val.tar",
-            },
-        )
-        for cluster, folder in standardized_torchvision_dataset_dirs.items()
+        cluster: Compose([
+            SymlinkDatasetFiles(
+                source=f"{datasets_folder}/imagenet",
+            ),
+            PrepareVisionDataset(tvd.ImageNet)
+        ])
+        for cluster, datasets_folder in standardized_torchvision_datasets_dirs.items()
+    },
+    tvd.KMNIST: {
+        cluster: Compose([
+            SymlinkDatasetFiles(
+                source=f"{datasets_folder}/kmnist",
+            ),
+            MoveFiles(
+                # Torchvision will look into a KMNIST/raw directory to
+                # preprocess the dataset
+                files=[("*", "KMNIST/raw/*")]
+            ),
+            PrepareVisionDataset(tvd.KMNIST)
+        ])
+        for cluster, datasets_folder in standardized_torchvision_datasets_dirs.items()
+    },
+    tvd.MNIST: {
+        # On the Mila and Beluga cluster we have archives which are extracted
+        # into 4 "raw" binary files. We do need to match the expected directory
+        # structure of the torchvision MNIST dataset though.  NOTE: On Beluga,
+        # we also have the MNIST 'raw' files in
+        # /project/rpp-bengioy/data/MNIST/raw, no archives.
+        cluster: Compose([
+            SymlinkDatasetFiles(
+                source=f"{datasets_folder}/mnist",
+            ),
+            MoveFiles(
+                # Torchvision will look into a raw directory to preprocess the
+                # dataset
+                files=[("*", "raw/*")]
+            ),
+            PrepareVisionDataset(tvd.MNIST)
+        ])
+        for cluster, datasets_folder in standardized_torchvision_datasets_dirs.items()
+    },
+    tvd.Places365: {
+        cluster: Compose([
+            SymlinkDatasetFiles(
+                source=f"{datasets_folder}/places365",
+            ),
+            SymlinkDatasetFiles(
+                source=f"{datasets_folder}/places365.var/places365_challenge",
+            ),
```

Comment:
These two are happening in the same directory? There's no issue with that?

Reply:
No the files will nicely merge